examples/DEVELOPER.md

```diff
@@ -0,0 +1,210 @@
+# Adding a sandbox example
+
+## Add a `verify.sh` test to your sandbox
+
+Sandboxes are tested as part of the continuous integration process, which expects
+each sandbox to have a `verify.sh` script containing tests for the example.
+
+### Basic layout of the `verify.sh` script
+
+At a minimum the `verify.sh` script should include the necessary parts to start
+and stop your sandbox.
+
+Given a sandbox with a single `docker` composition, adding the following
+to `verify.sh` will test that the sandbox can be started and stopped.
+
+```
+#!/bin/bash -e
+
+export NAME=example-sandbox
+
+# shellcheck source=examples/verify-common.sh
+. "$(dirname "${BASH_SOURCE[0]}")/../verify-common.sh"
+
+# add example tests here...
+
+```
+
+The `$NAME` variable is used for logging when testing the example, and will
+often be the same as the directory name.
+
+### Log before running each test
+
+There is a utility function `run_log` that can be used to indicate in test logs what
+is being executed and why, for example:
+
+```
+run_log "Checking foo.txt was created"
+ls foo.txt
+
+run_log "Checking bar.txt was created"
+ls bar.txt
+```
+
+### Add tests reflecting the documented examples
+
+The tests should follow the steps laid out in the documentation.
+
+For example, if the documentation provides a series of `bash` commands to execute, add these in order to `verify.sh`.
+
+You may wish to grep the response, or check the return code to ensure the command responds as expected.
+
+Likewise, if the documentation asks the user to browse to a page - eg http://localhost:8000 - then you should add a test
```

Comment:
e.g. or for example?

Reply:
not sure that e.g. is really more correct any more - there also seems to be a localisation angle on how its used. Personally i dont bother with dots for common abbreviations. Ill switch to "for example" to lessen cultural bias